tensorflow/src/TensorFlow/Internal/FFI.hs

```diff
@@ -187,8 +188,12 @@ createTensorData t = do
     -- Read data.
     len <- fromIntegral <$> Raw.tensorByteSize t
     bytes <- castPtr <$> Raw.tensorData t :: IO (Ptr Word8)
-    -- TODO(fmayle): Don't copy the data.
-    v <- S.fromList <$> peekArray len bytes
+    -- Note: We would like to avoid the copy by creating an S.Vector directly
+    -- from 'bytes' and calling Raw.deleteTensor when it gets GC'd, but we can't
+    -- because the vector may outlive the tensorflow session and become invalid.
```

Comment:
It sounds like what we'd like is to have SM.Vector with a custom destructor which we would hook up to call Raw.deleteTensor. Is this correct?